### PR TITLE
VITIS-11100 Display all AIE columns

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
#### Problem solved by the commit
The AIE report does not currently display any information for inactive columns which is very confusing. Additionally, the output JSON is very confusing.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Not a bug, this change improves the functionality of existing code.

#### How problem was solved, alternative solutions (if any) and why they were rejected
- Refactor how the AIE2 data is parsed in user space.
- Separate AIE and AIE2 reports. (This includes aie, aiemem, and aieshim)
- Change AIE2 json to output result of AIE2 parser instead AIE json.
- Refactor current AIE report to serve as a dispatcher to the AIE2 report.
- Inactive columns are now displayed, albeit with a inactive status.
- Hide very large items behind `--verbose` (locks and channels for AIE core, locks for AIE mem and for AIE shim)

#### Risks (if any) associated the changes in the commit
None.

#### What has been tested and how, request additional testing if necessary
Linux NPU
```
dbenusov@xlix-birman-29:/proj/rdi/staff/dbenusov/amd-aie-new$ xbutil examine -d -r aiemem -o test.json --force
1187: Created KMQ pcidev
334817: Device opened, fd=3
355001: Allocated KMQ BO for: userptr=0x7f11e0000000, size=50331648, flags=0x0
359750: Created KMQ device (0000:c3:00.1) ...
------------------------------------------------------------
                        EARLY ACCESS
        This release of xbutil contains early access
         experimental features which may have bugs.
------------------------------------------------------------

------------------------------
[0000:c3:00.1] : RyzenAI-npu1
------------------------------
AIE Mem Tiles
  Column 0
    Status: inactive
  Column 1
    Status: active
    Tiles
      Row 1
        DMA MM2S Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
        DMA S2MM Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
  Column 2
    Status: active
    Tiles
      Row 1
        DMA MM2S Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
        DMA S2MM Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
  Column 3
    Status: active
    Tiles
      Row 1
        DMA MM2S Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
        DMA S2MM Channels:
          Status  Queue Size  Queue Status  Current BD
          ----------------------------------------------
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
          Idle    0           okay          0
  Column 4
    Status: inactive
```
Snippet of JSON
```
{
    "schema_version": {
        "schema": "JSON",
        "creation_date": "Fri Mar 15 18:40:59 2024 GMT"
    },
    "devices": [
        {
            "interface_type": "pcie",
            "device_id": "0000:c3:00.1",
            "device_status": "UNKNOWN",
            "aie_mem": {
                "columns": [
                    {
                        "col": "0",
                        "status": "inactive"
                    },
                    ...
                    {
                        "col": "3",
                        "status": "active",
                        "tiles": [
                            {
                                "dma": {
                                    "mm2s_channels": [
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        }
                                    ],
                                    "s2mm_channels": [
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        },
                                        {
                                            "status": "Idle",
                                            "queue_size": "0",
                                            "queue_status": "okay",
                                            "current_bd": "0"
                                        }
                                    ]
                                },
                                "locks": [
                                    {
                                        "id": "0",
                                        "events": "0"
                                    },
                                    ...
                                    {
                                        "id": "63",
                                        "events": "0"
                                    }
                                ],
                                "row": "1"
                            }
                        ]
                    },
                    {
                        "col": "4",
                        "status": "inactive"
                    }
                ]
            }
        }
    ]
}
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by keryell** [comment]
> Additionally, the output JSON is very confusing.

I have nothing against you rewriting everything using YAML instead. ;-)

File: src/runtime_src/core/common/asd_parser.h [hunk 1]
```diff
@@ -53,6 +55,32 @@ struct aie_tiles_info
...
+  {
+    switch (tile_type) {
+      case aie_tile_type::core:
+        return core_rows;
+      case aie_tile_type::shim:
+        return shim_rows;
+      case aie_tile_type::mem:
+        return mem_rows;
+    }
+    throw std::runtime_error("Unknown tile type in formatting Aie tiles status info");
```
Comment (keryell) [inline src/runtime_src/core/common/asd_parser.h:69]:
```suggestion
    throw std::runtime_error("Unknown tile type in formatting AIE tiles status info");
```
in a few places

File: src/runtime_src/core/common/asd_parser.h [hunk 2]
```diff
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: Apache-2.0
-// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+// Copyright (C) 2023-2024 Advanced Micro Devices, Inc. All rights reserved.
 
-#ifndef _XCL_ASD_PARSER_H_
-#define _XCL_ASD_PARSER_H_
+#ifndef COMMON_AIE2_PARSER_H_
+#define COMMON_AIE2_PARSER_H_
```
Comment (stsoe) [inline src/runtime_src/core/common/asd_parser.h:5]: Huh? ASD?

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/common/asd_parser.h (hunk 2 above)
Comment (dbenusov-xilinx) [reply src/runtime_src/core/common/asd_parser.h:5]: Good catch I forgot to change the file name!

**Review by chvamshi-xilinx — APPROVED** [review]
Looks very good . 
Please check with Jason as they rely on the property tree populated from these reports.

**Review by jvillarre — APPROVED** [review]

**Pull request merged**